core/arch/arm/kernel/thread_a64.S

```diff
@@ -52,7 +74,15 @@ FUNC thread_resume , :
 	ldr	x0, [x0, THREAD_CTX_REGS_X0]
 	return_from_exception
 
-1:	load_xregs x0, THREAD_CTX_REGS_X1, 1, 3
+1:
+#ifdef	CFG_TA_PAUTH
+	/* Restore PAC keys before return to el0 */
+	load_xregs x0, THREAD_CTX_REGS_APIAKEY_HI, 1, 2
+	msr	S3_0_c2_c1_1, x1
```

Comment:
Please add read and write macros for these registers. See <out>/core/include/generated/arm32_sysreg.S for examples. In this case I guess we can hand code them and add them to core/arch/arm/include/arm64_macros.S.